From 864bfc5e0250141bb1da1644d416dc09dda308f5 Mon Sep 17 00:00:00 2001
From: Christine Gharzuzi <cgharzuzi@habana.ai>
Date: Tue, 04 May 2021 17:48:44 +0300
Subject: [PATCH] [SW-40455] gaudi: clear razwi event

Razwi event might occur during reset flow due
to access from host side to blocked RR

This commit clears and traces of razwi events
before uboot is loaded

Change-Id: Ibd3e138732755c245c80e586b3b5bbaf02a4d05e
Signed-off-by: Christine Gharzuzi <cgharzuzi@habana.ai>
---

diff --git a/gaudi/common.c b/gaudi/common.c
index 0a554a7..3e1367b 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -467,7 +467,7 @@
 }
 
 #define GAUDI_IRQ_SIZE	512
-static void preboot_ack_razwi_irq(void)
+void preboot_ack_razwi_irq(void)
 {
 	u32 i;
 
diff --git a/include/preboot.h b/include/preboot.h
index 89f0bf4..8656a91 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -168,6 +168,7 @@
 void preboot_finalize_enabled_caps(void);
 void preboot_conf_iatu_access(bool enable);
 void preboot_disable_iatu_access(void);
+void preboot_ack_razwi_irq(void);
 const char *preboot_get_version(enum preboot_vers ver);
 void hl_ram_init(void);
 void hl_pci_iatu_init(void);
diff --git a/preboot/module_uart.c b/preboot/module_uart.c
index 3f4d7ee..271f5f8 100644
--- a/preboot/module_uart.c
+++ b/preboot/module_uart.c
@@ -78,7 +78,17 @@
 	preboot_uart_rbuf_init();
 }
 
+__weak void preboot_ack_razwi_irq(void) {}
+
 void module_uart_teardown(void)
 {
 	preboot_uart_rbuf_flush();
+
+	/*
+	 * RAZWI event might appear during boot due to async access
+	 * from host side to locked range registers (PCI) (i.e. during
+	 * iatu initlization or reset flow, PCI access is disabled).
+	 * Therefore this ack is needed.
+	 */
+	preboot_ack_razwi_irq();
 }
